lc2fen/infer_pieces.py

```diff
@@ -196,52 +298,43 @@ def infer_chess_pieces(pieces_probs, a1_pos, previous_fen=None):
 
-    # Then set the blank spaces (the CNN has a very high accuracy of
-    # detecting these cases)
+    # Then identify the empty squares (the CNN has a very high accuracy of
+    # detecting empty squares)
     for idx, piece in enumerate(pieces_probs):
         if out_preds[idx] is None:
             if is_empty_square(piece):
                 out_preds[idx] = "_"
-                out_preds_empty -= 1
+                num_of_undetermined_squares -= 1
 
-    # Save if there is already a bishop in a [white, black] square
-    w_bishop_sq = [False, False]
-    b_bishop_sq = [False, False]
-
-    # Set the rest of the pieces in the order given by the highest
-    # probability of any piece for all the board
+    # Determine the locations of the other pieces in order of probability
+    # (there is a total of (`num_of_undetermined_squares` * 10) probabilities)
     pieces_lists = __sort_pieces_list(pieces_probs_sort)
-    # Index to the highest probability, from each list in pieces_lists,
-    # that we have not set yet (in the same order as above).
+    # Keep track of the indices to the squares, whose piece types have not been
+    # determined, with the highest probabilities in `pieces_lists` (there are 10
+    # piece types left, so we need to keep track of 10 indices)
     idx = [0] * 10
-    # Top of each sorted piece list (highest probability of each piece)
+    # Keep track of the top entry of each sorted piece list (corresponding to
+    # the square with the highest probability)
     tops = [piece_list[0] for piece_list in pieces_lists]
-    # Maximum number of pieces of each type in the same order as tops
+    # Maximum number of pieces of each type in the same order as `tops`
     max_pieces_left = [2, 2, 8, 2, 2, 2, 2, 8, 2, 2]
 
-    while out_preds_empty > 0:
-        # Fill in the square in out_preds that has the piece with the
-        # maximum probability of all the board
+    while num_of_undetermined_squares > 0:
+        # Determine the piece type of the square that has the piece with the
+        # highest probability across the entire board
         max_idx = __max_piece(tops)
         square = tops[max_idx][1]
-        # If we haven't maxed that piece type and the square is empty
+        # If we haven't maxed that piece type and the piece type of that square
+        # hasn't been determined, then we conclude that that square has exactly
+        # that piece
         if max_pieces_left[max_idx] > 0 and out_preds[square] is None:
-            # Fill the square and update counters
-            # If we have detected the move previously
-            if square == final_move_sq and possible_pieces:
-                # Only fill the square if one of the possible pieces
-                if __IDX_TO_PIECE[max_idx] in possible_pieces:
-                    out_preds[square] = __IDX_TO_PIECE[max_idx]
-                    out_preds_empty -= 1
-                    max_pieces_left[max_idx] -= 1
```

Comment:
The `square == final_move_sq and possible_pieces` condition is no longer needed because if `final_move_sq != -1` or `possible_pieces`, a move must've been successfully detected, so the FEN must've been concluded already